engine/class_modules/sc_death_knight.cpp

```diff
@@ -10067,6 +10149,9 @@ void death_knight_t::bone_shield_handler( const action_state_t* state ) const
   {
     buffs.bones_of_the_damned -> expire();
   }
+
+  // Value is stored in the spelldata as negative, so multiply by a positive
+  cooldown.dancing_rune_weapon -> adjust( 1.0 * legendary.crimson_rune_weapon -> effectN( 1).time_value() );
```

Comment:
```suggestion
  cooldown.dancing_rune_weapon -> adjust( legendary.crimson_rune_weapon -> effectN( 1 ).time_value() );
```
You don't need any special handling of the data here